src/electron/device-connect-view/components/device-connect-body.tsx

```diff
@@ -22,52 +23,24 @@ export type DeviceConnectBodyDeps = {
 
 export interface DeviceConnectBodyProps {
     deps: DeviceConnectBodyDeps;
+    viewState: DeviceConnectBodyState;
 }
 
-export interface DeviceConnectBodyState {
-    deviceConnectState: DeviceConnectState;
-    connectedDevice?: string;
-}
-
-export class DeviceConnectBody extends React.Component<DeviceConnectBodyProps, DeviceConnectBodyState> {
-    constructor(props: DeviceConnectBodyProps) {
-        super(props);
-        this.state = {
-            deviceConnectState: DeviceConnectState.Default,
-        };
-    }
-
-    public render(): JSX.Element {
-        const needsValidation = this.state.deviceConnectState !== DeviceConnectState.Connected;
-        const isConnecting = this.state.deviceConnectState === DeviceConnectState.Connecting;
-        const hasFailedConnecting = this.state.deviceConnectState === DeviceConnectState.Error;
-        const canStartTesting = this.state.deviceConnectState === DeviceConnectState.Connected;
-
-        return (
-            <div className="device-connect-body">
-                <DeviceConnectHeader />
-                <DeviceConnectPortEntry
-                    deps={this.props.deps}
-                    needsValidation={needsValidation}
-                    updateStateCallback={this.OnConnectedCallback}
-                />
-                <DeviceConnectConnectedDevice
-                    isConnecting={isConnecting}
-                    connectedDevice={this.state.connectedDevice}
-                    hasFailed={hasFailedConnecting}
-                />
-                <DeviceConnectFooter
-                    cancelClick={this.props.deps.currentWindow.close}
-                    canStartTesting={canStartTesting}
-                ></DeviceConnectFooter>
-            </div>
-        );
-    }
-
-    private OnConnectedCallback: UpdateStateCallback = (newState: DeviceConnectState, deviceName?: string) => {
-        this.setState({
-            deviceConnectState: newState,
-            connectedDevice: deviceName,
-        });
-    };
-}
+export const DeviceConnectBody = NamedFC<DeviceConnectBodyProps>('DeviceConnectBody', props => {
+    const isConnecting = props.viewState.deviceConnectState === DeviceConnectState.Connecting;
+    const hasFailedConnecting = props.viewState.deviceConnectState === DeviceConnectState.Error;
+    const canStartTesting = props.viewState.deviceConnectState === DeviceConnectState.Connected;
+
+    return (
+        <div className="device-connect-body">
+            <DeviceConnectHeader />
+            <DeviceConnectPortEntry deps={props.deps} viewState={{ deviceConnectState: props.viewState.deviceConnectState }} />
+            <DeviceConnectConnectedDevice
+                isConnecting={isConnecting}
+                connectedDevice={props.viewState.connectedDevice}
+                hasFailed={hasFailedConnecting}
```

Comment:
P2: (i know it is old code) it looks like DeviceConnectConnectedDevice can decide whether it is connecting / failed by reading the state props.viewState.deviceConnectState instead of the parent computing this value.

Reply:
Agreed, but:
- as you mention this PR is not changing `DeviceConnectConnectedDevice`
- I'd like to keep this PR as small as I can